cardinal/util.go

```diff
@@ -48,3 +52,34 @@ func isFatalError(err error) bool {
 	}
 	return true
 }
+
+func GetMessage[In any, Out any](wCtx WorldContext) (*MessageType[In, Out], error) {
```

Comment:
Keeping a note that this should be removed. Not sure why the previous comment is marked as resolved.

Reply:
Must have clicked it on accident. Will change this tomorrow.

Reply:
GetMessage gets a message type which provides a message id that can be used to add a transaction. The tests are doing this. 

EachMessage does not provide access to this id or the message type. 

```
	// Entities at index 5, 10 and 50 will be updated with some values
	modifyScoreMsg, err := cardinal.GetMessage[*ModifyScoreMsg, *EmptyMsgResult](wCtx)
	assert.NilError(t, err)
	tf.AddTransaction(
		modifyScoreMsg.ID(), &ModifyScoreMsg{
			PlayerID: ids[5],
			Amount:   105,
		},
	)
	tf.AddTransaction(
		modifyScoreMsg.ID(), &ModifyScoreMsg{
			PlayerID: ids[10],
			Amount:   110,
		},
	)
	tf.AddTransaction(
		modifyScoreMsg.ID(), &ModifyScoreMsg{
			PlayerID: ids[50],
			Amount:   150,
		},
	)
```

Reply:
The idiomatic way to handle this is to not have the test rely on GetMessage. Changing it to EachMessage doesn't fully work yet. So this isn't that straightforward of an edit. Again I suggest punting this to another ticket. This PR is getting far to large already. lmk if this works for you @smsunarto.

Reply:
fine [](edited-with-Axolo)